### DISCUSSION TITLE
Why /get_[api] /[api] pairs?

### DISCUSSION THREAD

**Discussion by venkatramanm** [beckn/protocol-specifications#229] [Q&A]
I see some apis like cancellation_reasons,return_reasons,rating_categories,feedback_categories,feedback_form, have  the pairing different from usual /[api] and /on_[api] paradigm !! Is there any reason for this ?

**Comment by ravi-prakash-v** [comment]
the "on_" prefix indicates the completion of an "event" like a search, or a select. Like if someone is searching for something using `search`, the `on_search` indicates that your `search` is over.. We have your results.

In the case of the meta APIs, it is not an "event" that is completed but rather a request for "data". Hence the prefix "get_" for requesting the data and no prefix for sending the _data_ that was requested.

That was my thought process. Maybe we can document this approach somewhere. Maybe CONTRIBUTION.md

**Reply by venkatramanm** [reply]
By this token, Is status call an event or a meta api? Is it not like get_status and status call?
